### PR TITLE
Clean up executor wrapping for tracing

### PR DESCRIPTION
**Pull request by sandorw**
**Goals (and why)**:
Enable http-remoting tracing by default on our executor services

**Implementation Description (bullets)**:
- Modify PTExecutors to auto-wrap
- Find existing executor services not using PTExecutors and switch them (I did not update instances in test code)

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:
PTExecutors.java

**Priority (whenever / two weeks / yesterday)**:

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2868)

### CODE REVIEW

**Comment by sandorw** [comment]
@carrino - I removed the 1.6 compatibility from the project with PTExecutors. Looking at https://github.com/palantir/atlasdb/issues/2111, is this still an issue? Would like to figure out a work around if this change will break our older projects.

File: commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java [hunk 1]
```diff
@@ -501,7 +487,13 @@ public void run() {
...
+        });
+        if (runnable instanceof Future<?>) {
+            @SuppressWarnings("unchecked")
+            Future<Object> unsafeFuture = (Future<Object>) runnable;
+            return new ForwardingRunnableFuture<Object>(wrappedRunnable, unsafeFuture);
+        }
+        return wrappedRunnable;
     }
 
     public static <T> RunnableFuture<T> wrap(RunnableFuture<T> rf) {
```
Comment (fsamuel-bs) [inline commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java]: I don't think we use this method, but let's also wrap it just to be safe.

**Comment by sandorw** [comment]
@carrino confirmed that it should be fine to drop the Java 6 compatibility here.

File: commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java (hunk 1 above)
Comment (sandorw) [reply commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java]: Thanks for catching that - fixed it and refactored the code a bit more.

**Pull request merged**